zoltan: Fixes for #40198

payerle opened the pull request:
fixes #40198

Fix issue in configure_args which resulted in duplicate "--with-ldflags" arguments (with different values) being passed to configure.  And extended the fix to similar arguments.

Also, repeated some flags to "--with-libs" to "--with-ldflags" as when the flags were only in "--with-libs", they did not seem to be picked up everywhere.  I suspect this is a bug in the configure script, but adding to both locations seems to solve it and should not have any adverse effects.

payerle committed:
zoltan: Fixes for #40198

Fix issue in configure_args which resulted in duplicate "--with-ldflags" arguments (with different values) being passed to configure.  And extended the fix to similar arguments.

Also, repeated some flags to "--with-libs" to "--with-ldflags" as when the flags were only in "--with-libs", they did not seem to be picked up everywhere.  I suspect this is a bug in the configure script, but adding to both locations seems to solve it and should not have any adverse effects.

alalazo commented on var/spack/repos/builtin/packages/zoltan/package.py:
```diff
@@ -143,19 +155,26 @@ def configure_args(self):
-        )
+        if config_cflags:
+            config_args.append("--with-cflags={0}".format(" ".join(config_cflags)))
+        if config_cxxflags:
+            config_args.append("--with-cxxflags={0}".format(" ".join(config_cflags)))
```
I think there is a typo here:
```suggestion
        if config_cxxflags:
            config_args.append("--with-cxxflags={0}".format(" ".join(config_cxxflags)))
```
Also, this can possibly be removed to save a copy of `cflags` and a couple of lines here:
```python
if config_cflags:
    config_args.append("--with-cflags={0}".format(" ".join(config_cflags)))
    config_args.append("--with-cxxflags={0}".format(" ".join(config_cflags)))
```

payerle committed:
Update var/spack/repos/builtin/packages/zoltan/package.py

Fixing typo caught by @alalazo

I prefer the more verbose version as it is more consistent and less likely to result in with-cxxflags argument being repeated.

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>
